website/source/guides/secret-mgmt/versioned-kv.html.md

```diff
@@ -0,0 +1,950 @@
+---
+layout: "guides"
+page_title: "Versioned KV Secret Engine - Guides"
+sidebar_current: "guides-secret-mgmt-versioned-kv"
+description: |-
+  Vault 0.10.0 introduced version 2 of key-value secret engine which supports
+  versioning of your secrets so that you can undo the accidental deletion of
+  secrets, or compare the different versions of the secret.
+---
+
+# Versioned Key/Value Secret Engine
+
+The [Static Secrets](/guides/secret-mgmt/static-secrets.html) guide introduced
+the basics of working with key-value secret engine. **Vault 0.10** introduced [_K/V
+Secrets Engine v2 with Secret
+Versioning_](https://www.hashicorp.com/blog/vault-0-10).  This guide
+demonstrates the new features introduced by the key-value secret engine v2 so
+that you can start leveraging them.
+
+
+## Reference Material
+
+- [Static Secrets guide](/guides/secret-mgmt/static-secrets.html)
+- [KV Secrets Engine - Version 2](/docs/secrets/kv/kv-v2.html)
+- [KV Secrets Engine - Version 2 (API)](/api/secret/kv/kv-v2.html)
+
+## Estimated Time to Complete
+
+10 minutes
+
+
+## Challenge
+
+The KV secret engine v1 does not provide a way to version or roll back secrets.
+This made it difficult to recover from unintentional data loss or overwrite when
+more than one user is writing at the same path.
+
+
+## Solution
+
+Run the **version 2** of KV secret engine which can retain a configurable
+number of secret versions. This enables older versions' data to be retrievable
+in case of unwanted deletion or updates of the data.  In addition, its
+_Check-and-Set_ operations can be used to protect the data from overwritten
+unintentionally.
+
+![Versioned KV](/assets/images/vault-versioned-kv-1.png)
+
+## Prerequisites
+
+To perform the tasks described in this guide, you need to have a Vault
+environment.  Refer to the [Getting
+Started](/intro/getting-started/install.html) guide to install Vault. Make sure
+that your Vault server has been [initialized and
+unsealed](/intro/getting-started/deploy.html).
+
+### Policy requirements
+
+-> **NOTE:** For the purpose of this guide, you can use **`root`** token to work
+with Vault. However, it is recommended that root tokens are only used for just
+enough initial setup or in emergencies. As a best practice, use tokens with
```

Comment:
However, it is recommended that root tokens are only used for initial setup or in emergencies